src/components/TextFieldComp.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { TextField } from '@material-ui/core';
+import PropTypes from 'prop-types';
+
+function TextFieldComp(props) {
+  const { id, label, value } = props;
+
+  function handleFieldChange(e) {
+    props.onChange(e.target.id, e.target.value);
+  }
+  return (
+    <TextField
+      id={id}
+      label={label}
+      type="text"
+      value={value}
+      onChange={handleFieldChange}
+      fullWidth
+    />
+  );
```

Comment:
This could all be simplified to:
```suggestion
  return (
    <TextField {...props} fullWidth/>
  );
```
At which point I don't believe a separate function component is worth it.